actors/builtin/market/market_state.go

```diff
@@ -149,9 +149,6 @@ func (m *marketStateMutation) updatePendingDealState(rt Runtime, state *DealStat
 	}
 
 	nextEpoch = epoch + DealUpdatesInterval
-	if nextEpoch > deal.EndEpoch {
-		nextEpoch = deal.EndEpoch
-	}
```

Comment:
Thanks. Please leave a comment here stating that we're explicitly not inspecting the end epoch and may process a deal's expiration late, in order to avoid outside influence over cron loading.